baseband/mark5b/base.py

```diff
@@ -258,14 +258,15 @@ def read(self, count=None, fill_value=0., out=None):
 
         return out
 
-    def _read_frame(self):
+    def _read_frame(self, fill_value=0.):
         self.fh_raw.seek(self.offset // self.samples_per_frame *
                          self._frame.size)
         self._frame = self.read_frame(ref_mjd=self.header0.kday,
                                       nchan=self._sample_shape.nchan,
                                       bps=self.bps)
         # Convert payloads to data array.
         self._frame_data = self._frame.data
+        self._frame.invalid_data_value = fill_value
```

Comment:
Same as for mark4 (both moving this up and using `self._frame_data` in `read()`)